### PR TITLE
Dry-run JSON debug output 6

### PR DESCRIPTION
**Pull request by isseis**
This pull request enhances the dry-run debug JSON output implementation plan and detailed design with a strong focus on security and performance. It introduces rigorous error handling to prevent information leakage and log injection, refines the test cases to cover security edge cases, and expands the documentation and implementation plan with precise performance measurement and optimization strategies.

**Security Improvements**

* Updated the `UnmarshalJSON` error handling for `InheritanceMode` to avoid including input values in error messages, mitigating risks of log injection and information disclosure. Additional comments clarify the importance of this validation layer for future JSON config support. [[1]](diffhunk://#diff-1ad3e3c89353c17e51b918f577e20d53c15a3aef1143f9ce45f2cabd4e1b4d6bR174-R187) [[2]](diffhunk://#diff-1ad3e3c89353c17e51b918f577e20d53c15a3aef1143f9ce45f2cabd4e1b4d6bL191-R216)
* Added comprehensive security-focused test cases for `InheritanceMode` deserialization, including invalid values, log injection attempts, path traversal, and empty strings. Tests assert that error messages do not leak sensitive input. [[1]](diffhunk://#diff-1ad3e3c89353c17e51b918f577e20d53c15a3aef1143f9ce45f2cabd4e1b4d6bR1020-R1101) [[2]](diffhunk://#diff-1ad3e3c89353c17e51b918f577e20d53c15a3aef1143f9ce45f2cabd4e1b4d6bR1112-R1115)
* Updated implementation plan and checklist to explicitly require and verify that error messages do not contain input values, and to include security tests for log injection and path traversal. [[1]](diffhunk://#diff-1530741f122f63d88804d01ec0cd8ddcf3967988f0551c829814ff115ef00699R129-R134) [[2]](diffhunk://#diff-1530741f122f63d88804d01ec0cd8ddcf3967988f0551c829814ff115ef00699R151-R162)

**Performance Measurement and Optimization**

* Expanded the performance goals section with detailed measurement methods for execution time and memory usage, including benchmarking scripts, configuration file variations, profiling tools, and acceptance criteria for overhead.
* Revised the implementation plan to include step-by-step benchmarks for various configuration sizes, overhead calculations, and memory profiling, with increased time estimates to reflect the more thorough testing and optimization. [[1]](diffhunk://#diff-1530741f122f63d88804d01ec0cd8ddcf3967988f0551c829814ff115ef00699L880-R922) [[2]](diffhunk://#diff-1530741f122f63d88804d01ec0cd8ddcf3967988f0551c829814ff115ef00699L914-R963)

**Logging and Output Separation**

* Clarified the separation between dry-run output and error logging, ensuring sensitive information is only revealed when explicitly requested and is never included in logs. Provided code examples and updated documentation for safe logging practices. [[1]](diffhunk://#diff-1ad3e3c89353c17e51b918f577e20d53c15a3aef1143f9ce45f2cabd4e1b4d6bL545-R568) [[2]](diffhunk://#diff-1ad3e3c89353c17e51b918f577e20d53c15a3aef1143f9ce45f2cabd4e1b4d6bL1202-R1475)

**Test Coverage and Documentation**

* Added and documented new test cases for environment variable masking and detail level handling, ensuring correct behavior for sensitive data and output formats. [[1]](diffhunk://#diff-1ad3e3c89353c17e51b918f577e20d53c15a3aef1143f9ce45f2cabd4e1b4d6bR868-R869) [[2]](diffhunk://#diff-1ad3e3c89353c17e51b918f577e20d53c15a3aef1143f9ce45f2cabd4e1b4d6bR879-R880) [[3]](diffhunk://#diff-1ad3e3c89353c17e51b918f577e20d53c15a3aef1143f9ce45f2cabd4e1b4d6bR890-R893) [[4]](diffhunk://#diff-1ad3e3c89353c17e51b918f577e20d53c15a3aef1143f9ce45f2cabd4e1b4d6bL861-R922) [[5]](diffhunk://#diff-1ad3e3c89353c17e51b918f577e20d53c15a3aef1143f9ce45f2cabd4e1b4d6bL884-R948) [[6]](diffhunk://#diff-1ad3e3c89353c17e51b918f577e20d53c15a3aef1143f9ce45f2cabd4e1b4d6bR747-R748) [[7]](diffhunk://#diff-1ad3e3c89353c17e51b918f577e20d53c15a3aef1143f9ce45f2cabd4e1b4d6bR764-R766) [[8]](diffhunk://#diff-1ad3e3c89353c17e51b918f577e20d53c15a3aef1143f9ce45f2cabd4e1b4d6bL757-R799) [[9]](diffhunk://#diff-1ad3e3c89353c17e51b918f577e20d53c15a3aef1143f9ce45f2cabd4e1b4d6bL785-R830) [[10]](diffhunk://#diff-1ad3e3c89353c17e51b918f577e20d53c15a3aef1143f9ce45f2cabd4e1b4d6bR983-R998)
* Updated time estimates and documentation to reflect the additional security and performance work.

These changes collectively ensure that the dry-run debug JSON output feature is robust, secure, and performant, with clear test coverage and documentation supporting both implementation and future maintenance.

### CODE REVIEW

File: internal/runner/runnertypes/config_test.go [hunk 1]
```diff
@@ -277,6 +301,19 @@ func TestInheritanceMode_UnmarshalJSON(t *testing.T) {
...
+					errMsg := err.Error()
+					// Extract the input value without JSON quotes for checking
+					var inputValue string
+					_ = json.Unmarshal([]byte(tt.input), &inputValue)
+					// Skip empty string check (would be a false positive)
+					if inputValue != "" {
+						// The error message should only contain the base error, not the input
+						assert.NotContains(t, errMsg, inputValue,
+							"error message should not contain user input to prevent log injection")
+					}
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/runnertypes/config_test.go]: done
